source/httpfs/driver.go

```diff
@@ -10,10 +10,21 @@ import (
 	"github.com/golang-migrate/migrate/v4/source"
 )
 
-// driver is a migration source driver for reading migrations from
+// ErrDuplicateMigration is an error type for reporting duplicate migration
+// files.
+type ErrDuplicateMigration struct {
```

Comment:
Move this up one package level e.g. place it in the `source` package. I think a new `errors.go` file would be nice.